src/sarif-converter-21.ts

```diff
@@ -102,6 +108,16 @@ export class SarifConverter21 {
         return run;
     }
 
+    private convertResultsToRules(results: DecoratedAxeResults) {
+        const resultToRuleConverter: ResultToRuleConverter = new ResultToRuleConverter(
+            results,
+            this.wcagLinkDataIndexer.getSortedWcagTags(),
+            this.wcagLinkDataIndexer.getWcagTagsToTaxaIndices(),
+        );
+        this.ruleIdsToRuleIndices = resultToRuleConverter.getRuleIdsToRuleIndices();
```

Comment:
I wouldn't have expected this side effect based on the function name. Generally, I would recommend against doing this sort of field initialization in a non-obvious place; it means that the order of calls to this function vs other functions using ruleIdsToRuleIndices is fixed/important in a way that wouldn't be obvious to someone reading the callsite for this function.

Can you try using parameters to pass the necessary data around, instead of the field?